src/pulse3D/plate_recording.py

```diff
@@ -382,11 +383,16 @@ def __init__(
         # set up platemap info
         self.platemap_name = self.wells[0][PLATEMAP_NAME_UUID]
 
-        platemap_labels = defaultdict(list)
-        for well_file in self:
-            label = well_file[PLATEMAP_LABEL_UUID]
-            platemap_labels[label].append(well_file[WELL_NAME_UUID])
-        self.platemap_labels = dict(platemap_labels)
+        if well_groups is None:
+            platemap_labels = defaultdict(list)
+            for well_file in self:
+                label = well_file[PLATEMAP_LABEL_UUID]
+                if label != NOT_APPLICABLE_LABEL:
+                    platemap_labels[label].append(well_file[WELL_NAME_UUID])
+        else:
+            platemap_labels = defaultdict(**well_groups)
+
+        self.platemap_labels = platemap_labels
```

Comment:
We should still convert this back to a normal dict to avoid weird bugs by using accidentally accessing a key that doesn't exist